ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -924,13 +966,14 @@ int get_accumulate_req(const void *origin_addr, int origin_count,
         return ret;
     }
 
-    ret = opal_common_ucx_ctx_flush(module->ctx, OPAL_COMMON_UCX_SCOPE_EP, target);
-    if (ret != OMPI_SUCCESS) {
-        return ret;
-    }
-
     if (op != &ompi_mpi_op_no_op.op) {
         if (op == &ompi_mpi_op_replace.op) {
+            ret = opal_common_ucx_ctx_flush(module->ctx, OPAL_COMMON_UCX_SCOPE_EP,
```

Comment:
Just curious: is that flush required here? Would it be possible to fence before the put instead and possibly avoid blocking on the flush?

Reply:
Yes, this would be possible. Actually, we are doing the same thing in the proposed nonblocking version of the get-accumulate.